File: easydoc-javadoc/src/main/java/com/star/easydoc/view/template/AbstractTemplateConfigurable.java
package com.star.easydoc.view.template;

import javax.swing.*;

import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.options.Configurable;
import com.star.easydoc.config.EasyJavadocConfigComponent;
import com.star.easydoc.model.EasyJavadocConfiguration;
import org.jetbrains.annotations.Nullable;

/**
 * @author <a href="mailto:wangchao.star@gmail.com">wangchao</a>
 * @version 1.0.0
 * @since 2019-11-10 18:14:00
 */
public abstract class AbstractTemplateConfigurable implements Configurable {

    protected EasyJavadocConfiguration config = ServiceManager.getService(EasyJavadocConfigComponent.class).getState();

    @Nullable
    @Override
    public JComponent createComponent() {
        return getView().getComponent();
    }

    public abstract AbstractTemplateConfigView getView();
}

File: easydoc-kdoc/src/main/kotlin/com/star/easydoc/kdoc/view/template/AbstractTemplateConfigurable.kt
package com.star.easydoc.kdoc.view.template

import com.intellij.openapi.options.Configurable
import javax.swing.JComponent

/**
 * @author [wangchao](mailto:wangchao.star@gmail.com)
 * @version 1.0.0
 * @since 2019-11-10 18:14:00
 */
abstract class AbstractTemplateConfigurable : Configurable {

    override fun createComponent(): JComponent {
        return view.component
    }

    abstract val view: AbstractTemplateConfigView
}